### PR TITLE
[libc][NFC] Move functions from `FPBits` to `FPRep`, make `bits` member private

### PR DESCRIPTION
**Pull request by gchatelet**

### CODE REVIEW

**Comment by gchatelet** [comment]
We can probably rewrite the code that directly manipulates the bit representation with higher level functions.

File: libc/src/__support/FPUtil/FPBits.h [hunk 1]
```diff
@@ -652,6 +655,44 @@ struct FPRep : public FPRepSem<fp_type, RetT> {
     bits = merge(bits, mantVal, FRACTION_MASK);
   }
 
+  // TODO: Use an uint32_t for 'biased_exp'.
```
Comment (legrosbuffle) [inline libc/src/__support/FPUtil/FPBits.h:658]: s/TODO/FIXME

File: libc/src/__support/FPUtil/FPBits.h [hunk 2]
```diff
@@ -652,6 +655,44 @@ struct FPRep : public FPRepSem<fp_type, RetT> {
...
+  LIBC_INLINE static constexpr RetT
+  create_value(Sign sign, StorageType biased_exp, StorageType mantissa) {
+    static_assert(fp_type != FPType::X86_Binary80,
+                  "This function is not tested for X86 Extended Precision");
+    return RetT(encode(sign, BiasedExp(static_cast<uint32_t>(biased_exp)),
+                       Sig(mantissa)));
+  }
+
+  // The function convert integer number and unbiased exponent to proper float
```
Comment (legrosbuffle) [inline libc/src/__support/FPUtil/FPBits.h:667]: *converts

File: libc/src/__support/FPUtil/FPBits.h [hunk 3]
```diff
@@ -652,6 +655,44 @@ struct FPRep : public FPRepSem<fp_type, RetT> {
     bits = merge(bits, mantVal, FRACTION_MASK);
   }
 
+  // TODO: Use an uint32_t for 'biased_exp'.
+  LIBC_INLINE static constexpr RetT
+  create_value(Sign sign, StorageType biased_exp, StorageType mantissa) {
```
Comment (legrosbuffle) [inline libc/src/__support/FPUtil/FPBits.h:660]: Doc ?

File: libc/src/__support/FPUtil/FPBits.h [hunk 4]
```diff
@@ -652,6 +655,44 @@ struct FPRep : public FPRepSem<fp_type, RetT> {
...
+                  "This function is not tested for X86 Extended Precision");
+    return RetT(encode(sign, BiasedExp(static_cast<uint32_t>(biased_exp)),
+                       Sig(mantissa)));
+  }
+
+  // The function convert integer number and unbiased exponent to proper float
+  // T type:
+  //   Result = number * 2^(ep+1 - exponent_bias)
+  // Be careful!
+  //   1) "ep" is raw exponent value.
```
Comment (legrosbuffle) [inline libc/src/__support/FPUtil/FPBits.h:671]: *is the

File: libc/src/__support/FPUtil/FPBits.h [hunk 5]
```diff
@@ -652,6 +655,44 @@ struct FPRep : public FPRepSem<fp_type, RetT> {
...
+  //   2) The function add to +1 to ep for seamless normalized to denormalized
```
Comment (legrosbuffle) [inline libc/src/__support/FPUtil/FPBits.h:672]: *adds

File: libc/src/__support/FPUtil/FPBits.h [hunk 6]
```diff
@@ -652,6 +655,44 @@ struct FPRep : public FPRepSem<fp_type, RetT> {
...
+  //      transition.
+  //   3) The function did not check exponent high limit.
```
Comment (legrosbuffle) [inline libc/src/__support/FPUtil/FPBits.h:674]: *does

**Review by legrosbuffle — APPROVED** [review]

**Pull request merged**